x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/audit/logfile/CapturingLogger.java

```diff
@@ -59,28 +65,38 @@ public static boolean isEmpty(final String name) {
         public final List<String> trace = new ArrayList<>();
 
         private MockAppender(final String name) throws IllegalAccessException {
-            super(name, RegexFilter.createFilter(".*(\n.*)*", new String[0], false, null, null), null);
+            this(name, null);
+        }
+
+        private MockAppender(final String name, StringLayout layout) throws IllegalAccessException {
+            super(name, RegexFilter.createFilter(".*(\n.*)*", new String[0], false, null, null), layout);
         }
 
         @Override
         public void append(LogEvent event) {
+            final String logLine;
+            if (getLayout() != null) {
+                logLine = new String(getLayout().toByteArray(event), StandardCharsets.UTF_8);
+            } else {
+                logLine = event.getMessage().getFormattedMessage();
+            }
```

Comment:
Use a layout, if provided, when printing the event. Assumes the layout encodes text as UTF8.